packages/template-retail-react-app/app/pages/checkout/partials/payment-form.jsx

```diff
@@ -0,0 +1,95 @@
+ */
+import React from 'react'
+import {FormattedMessage, FormattedNumber, useIntl} from 'react-intl'
+import PropTypes from 'prop-types'
+import {Box, Flex, Radio, RadioGroup, Stack, Text, Tooltip} from '@chakra-ui/react'
+import {useCurrentBasket} from '../../../hooks/use-current-basket'
+import {LockIcon, PaypalIcon} from '../../../components/icons'
+import CreditCardFields from '../../../components/forms/credit-card-fields'
+
+const PaymentForm = ({form, onSubmit}) => {
+    const {formatMessage} = useIntl()
+    const {data: basket} = useCurrentBasket()
+
+    return (
+        <form onSubmit={form.handleSubmit(onSubmit)}>
+            <Stack spacing={8}>
+                <Stack spacing={5}>
+                    <Box border="1px solid" borderColor="gray.100" rounded="base" overflow="hidden">
+                        <RadioGroup value="cc">
+                            <Box
+                                py={3}
+                                px={[4, 4, 6]}
+                                bg="gray.50"
+                                borderBottom="1px solid"
+                                borderColor="gray.100"
+                            >
+                                <Radio value="cc">
+                                    <Flex justify="space-between">
+                                        <Stack direction="row" align="center">
+                                            <Text fontWeight="bold">
+                                                <FormattedMessage
+                                                    defaultMessage="Credit Card"
+                                                    id="payment_selection.heading.credit_card"
+                                                />
+                                            </Text>
+                                            <Tooltip
+                                                hasArrow
+                                                placement="top"
+                                                label={formatMessage({
+                                                    defaultMessage:
+                                                        'This is a secure SSL encrypted payment.',
+                                                    id: 'payment_selection.tooltip.secure_payment'
+                                                })}
+                                            >
+                                                <LockIcon color="gray.700" boxSize={5} />
+                                            </Tooltip>
+                                        </Stack>
+                                        <Text fontWeight="bold">
+                                            <FormattedNumber
+                                                value={basket.orderTotal}
+                                                style="currency"
+                                                currency={basket.currency}
```

Comment:
Should this follow the pattern of `currency={currency || basket?.currency || activeCurrency}`? Taken from another file..

Reply:
I'm currently stuck in the checkout flow. Can't get pass the Shipping step.

Seeing this error message:
![Google Chrome 2023-03-22 at 16 32 33](https://user-images.githubusercontent.com/847300/227062068-933c10aa-e67a-41f5-a675-18159b5bcee2.png)

Reply:
oops, `const {currency} = useCurrency` forgot parentheses...